spec/services/suri_service_spec.rb

```diff
@@ -52,6 +52,9 @@
 
     before :each do
       @mock_repo = double(Rubydora::Repository)
+      @mock_client = double(Rubydora::RestApiClient)
```

Comment:
what about turning these into `instance_double` calls, instead of regular `double` calls?  would that have caught the problem with the bad mocking before, since `instance_double` requires `expect`/`allow` mocked method calls to exist on the class being mocked?